### ISSUE TITLE
Call for help: No longer able to open meeting invitations in app

### ISSUE THREAD

**Issue by mbert** [IsmaelMartinez/teams-for-linux#675]
It looks like Microsoft have changed their teams.microsoft.com site's behaviour: while before, when opening a meeting invitation in a browser one would get prompted to either open in the browser or in the app, the site now immediately redirects into the teams webapp in the browser, no way to open the meeting in the app. 

![teams-screenshot](https://user-images.githubusercontent.com/383271/216048308-97bb1ef0-d544-4793-bff0-7af949aff917.png)

The above screenshot shows what I can see for a few milliseconds before the redirect takes place (and nothing on the right side is linked to anything). The native dialogue asking me which app I want to open the link with no longer appears.

All my Linux-using colleagues have had the same problem since this morning.

This is of course not a problem of teams-for-linux, but since MS may not be willing to change this (annoying) behaviour, I'd like to ask around whether anyone has found a workaround for this?

[edit] This behaviour is triggered by a "Linux" in the user agent.

**Comment by mbert** [comment]
I have found a workaround that does the job for now:

Extension **User-Agent Switcher for Chrome** (I am using a Chromium-based browser).
It allows configuring User-Agent-Spoofing by site. I have set up **Firefox 33 for Windows** whenever I open teams.microsoft.com, after that I got my "open" dialogue back.

**Comment by scristofaro** [comment]
> I have found a workaround that does the job for now:
> 
> Extension **User-Agent Switcher for Chrome** (I am using a Chromium-based browser). It allows configuring User-Agent-Spoofing by site. I have set up **Firefox 33 for Windows** whenever I open teams.microsoft.com, after that I got my "open" dialogue back.

That's a workaround, but i think it's should better force teams-for-linux to open teams link directly in app, if possible.

**Comment by EnricoDamini** [comment]
Thanks @mbert same issue for me but spoffing user agent not work

**Comment by norman-knott** [comment]
Experiencing the same Problem when using Firefox in Ubuntu. But the not usable choice screen looks different to me.

![Screenshot from 2023-02-07 09-15-35](https://user-images.githubusercontent.com/18596176/217202993-8c1b8952-43b4-465f-bcc7-525f9a35628e.png)

Before, when it worked to open a meeting in the teams-for-linux app, it looked like the screen in the issue description.

**Comment by mirenradia** [comment]
A temporary workaround would be the ability to paste a meeting link somewhere in this app that will then be navigated to.

**Comment by mbert** [comment]
> A temporary workaround would be the ability to paste a meeting link somewhere in this app that will then be navigated to.

I second that. However I am not sure whether this is easily technically possible, because the application is merely a web wrapper.

**Comment by IsmaelMartinez** [comment]
It use to work pasting it on a chat and clicking. See if you can call teams-for-linux with the --open-url config option (it might have a different name, check the config options). Ms strikes again!!

**Comment by IsmaelMartinez** [comment]
There is some code that use to "click" the open in browser option a while back. If someone finds that selector we can try to fix that again... But ms will change it again soon. (I am sure)

**Comment by munen** [comment]
We had the same issue with Teams. Changing the UA did not help. However, we found an easy way to get teams working, again. Here’s how we did it: https://200ok.ch/posts/2023-02-07_make_teams_work_again.html

**Comment by norman-knott** [comment]
@munen For teams-for-linux, at my PC, this alias simply tries to open the app if it is not already open. If it is already open, sometimes it reloads the content. But unfortunately it does not open the call.
Are you using the original MS Teams Desktop client?

**Comment by munen** [comment]
> Are you using the original MS Teams Desktop client?

No. That wonderful piece of software never worked for me. I’m using Teams from this Repo.

> For teams-for-linux, at my PC, this alias simply tries to open the app if it is not already open.

Teams did that to me as well some months ago. I was not able to join calls, because links didn’t open any calls. It does work for me in the latest version, though.

**Comment by mbert** [comment]
It does not work here either.  I have tried to track things down. Even if I do an 'xdg-open msteams://....' link from the console, my teams-for-linux window gets the focus, but it does not open the meeting link.

**Comment by munen** [comment]
Opening Teams via the regular link stopped working for both me and @branch14 from the very same meeting and that the documented fix in the blog post works for both of us. 

Sorry to hear that it doesn’t work for everyone. Maybe you have better luck in switching all meetings to software that actually works and doesn’t Bork on trivial stuff like opening links. Good luck to everyone🙏🏻

Having said so, a huge thank to to the maintainers of this repository to making Teams at least available on Linux🙇‍♂️

**Comment by branch14** [comment]
@mbert No matter how you call it (xdg-open, teams, teams-for-linux, or the proposed alias) you have to make sure the URL is quoted. Because the contains characters like `&` that will be interpreted by your shell if unquoted.

```
xdg-open "msteams:/..."
```

**Comment by mbert** [comment]
> @mbert No matter how you call it (xdg-open, teams, teams-for-linux, or the proposed alias) you have to make sure the URL is quoted. Because the contains characters like `&` that will be interpreted by your shell if unquoted.
> 
> ```
> xdg-open "msteams:/..."
> ```

Sure, I did so. But it simply focuses the teams-for-linux window.

I actually don't need it, because the variant with the spoofed UA is more convenient. I click the link, it opens a tab in the browser, the dialog comes, and I can continue in the app.

**Comment by csonuryilmaz** [comment]
> A temporary workaround would be the ability to paste a meeting link somewhere in this app that will then be navigated to.

Thanks @mirenradia :hugs: it's the only working workaround for me.

**Comment by hightowe** [comment]
I could not get the bash alias to work for me, that is presented here: https://200ok.ch/posts/2023-02-07_make_teams_work_again.html

And so I wrote this little program that does the same thing, and does work for me:

~~~
hightowe@zion ~ $ cat ~/bin/teams4linux
#!/bin/bash

###############################################################
# Quick xdg-open launcher script that was inspired by
# https://200ok.ch/posts/2023-02-07_make_teams_work_again.html
# which I could not get to work on my bash...
#
# Written 2023-Feb-20 by Lester Hightower
###############################################################

# The BASE_URL to replace with msteams:
BASE_URL='https://teams.microsoft.com'

url="$1"

echo "$url" | egrep -q -e "^$BASE_URL"
if [ $? -ne 0 ]; then
  echo "Invalid url vs expected ($BASE_URL): $url"
  exit -1
fi

path=$(echo "$url" | sed -e  "s%^$BASE_URL%%")
if [ $? -ne 0 ]; then
  echo "Failed to find path with sed"
  exit -1
fi

new_url="msteams:$path"

echo -e "RUNNING:\n"xdg-open "'$new_url'"

# Uncomment the next line if you want xdg-open to be run
#xdg-open "$new_url"
~~~

Hope it helps someone...

**Comment by MortenVinding** [comment]
> I could not get the bash alias to work for me, that is presented here: https://200ok.ch/posts/2023-02-07_make_teams_work_again.html

It's missing a semicolon :)
Should be:

alias pott='f() { xdg-open ${1/https:\/\/teams.microsoft.com/msteams:}; }; f'
instead of:
alias pott='f() { xdg-open ${1/https:\/\/teams.microsoft.com/msteams:} }; f'

**Comment by hightowe** [comment]
@MortenVinding - thanks. However, I still don't get the result that I would expect... Example:

~~~
hightowe@zion ~ $ alias pott
bash: alias: pott: not found
hightowe@zion ~ $ alias pott='f() { echo ${1/https://teams.microsoft.com/msteams:}; }; f'
hightowe@zion ~ $ pott https://teams.microsoft.com/l/foo/bar
/teams.microsoft.com/msteams://teams.microsoft.com/l/foo/bar
hightowe@zion ~ $ 
~~~

I expect simply: msteams:/l/foo/bar

**Comment by MortenVinding** [comment]
> However, I still don't get the result that I would expect...

sorry somehow the escaping got removed. should have been:
alias pott='f() { xdg-open ${1/https:\/\/teams.microsoft.com/msteams:}; }; f'

**Comment by hightowe** [comment]
@MortenVinding - Looks like the escaping was removed again. To keep it, you must use a code snippet here.

Here is what I think it should be:

~~~
alias pott='f() { xdg-open ${1/https:\/\/teams.microsoft.com/msteams:}; }; f'
~~~

A test of that using echo worked for me:

~~~
hightowe@zion ~ $ alias pott='f() { echo ${1/https:\/\/teams.microsoft.com/msteams:}; }; f'
hightowe@zion ~ $ pott https://teams.microsoft.com/l/foo/bar
msteams:/l/foo/bar
~~~

Thanks again.

**Comment by MortenVinding** [comment]
> Looks like the escaping was removed again. To keep it, you must use a code snippet here.

Yes I see. But good you found out anyway.
Your welcome. Let hope we soon get a better solution.

**Comment by mirenradia** [comment]
Has anyone figured out how to get it to open in a specific (non-default) profile (#690)? I often have meetings with an organization I have a guest account in which is not my main one.

**Comment by DerXteMensch** [comment]
@munen @hightowe How can I use your program or the alias? Just have no idea how to use it. Can it be used automatically and permanent?

**Comment by IsmaelMartinez** [comment]
I suspect then that the logic that needs tweak is around this area:

https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/app/mainAppWindow/index.js#L156

That should handle the meeting calls and the redirect (in principle, but it might be now broken)

**Comment by nxadm** [comment]
> A temporary workaround would be the ability to paste a meeting link somewhere in this app that will then be navigated to.

That would be rather a solution for me. Less things that possible can break.

**Comment by IsmaelMartinez** [comment]
The changes from https://github.com/IsmaelMartinez/teams-for-linux/pull/713 might solve this issue

**Comment by IsmaelMartinez** [comment]
pre-release in https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.0.56

**Comment by nxadm** [comment]
@IsmaelMartinez I tried the pre-release, but it does not solved. I ended workarounding it with a Firefox addon ([Redirector](https://addons.mozilla.org/en-US/firefox/addon/redirector/)) with the attached configuration.
![image](https://user-images.githubusercontent.com/3687879/227589173-4a4357c1-07b9-4c35-b5d0-457b0e017605.png)

**Comment by mirenradia** [comment]
> @IsmaelMartinez I tried the pre-release, but it does not solved. I ended workarounding it with a Firefox addon ([Redirector](https://addons.mozilla.org/en-US/firefox/addon/redirector/)) with the attached configuration. ![image](https://user-images.githubusercontent.com/3687879/227589173-4a4357c1-07b9-4c35-b5d0-457b0e017605.png)

Thanks for this workaround. IMO, this works better for me than the shell script/function/alias.

It would still be great to be able to redirect it to specific profiles.

**Comment by jijojosephk** [comment]
Is there a way to bypass the default browser behaviour? or at least change the default browser for certain urls?

**Comment by jijojosephk** [comment]
> Experiencing the same Problem when using Firefox in Ubuntu. But the not usable choice screen looks different to me.
> 
> ![Screenshot from 2023-02-07 09-15-35](https://user-images.githubusercontent.com/18596176/217202993-8c1b8952-43b4-465f-bcc7-525f9a35628e.png)
> 
> Before, when it worked to open a meeting in the teams-for-linux app, it looked like the screen in the issue description.

I guess this is because website understands that the host is linux and no desktop app for linux so use only web app.

**Comment by jijojosephk** [comment]
Since app has no solution, I'm closing but will pin this.

**Comment by jovandeginste** [comment]
Is there a working solution now to this issue? I tried all suggestions, but I still can't join a meeting via link. The teams app "opens" the URL, but does nothing with it...

**Comment by DerXteMensch** [comment]
The solution from @nxadm using a Firefox-addon works.

**Comment by jovandeginste** [comment]
> The solution from @nxadm using a Firefox-addon works.

I'm using that, but to no avail... My workaround is to open the meeting on my phone, type something in the chat, which then pops up the chat on my desktop, from where I can join the call. That's just a little workaround, I know, but I would appreciate a better way :wink:

**Comment by jovandeginste** [comment]
![image](https://user-images.githubusercontent.com/3170771/236403948-d5da915b-7057-472f-9b27-1c07cd6bbbb0.png)

**Comment by zendynar** [comment]
> A temporary workaround would be the ability to paste a meeting link somewhere in this app that will then be navigated to.
This is what I do, I have a chat with myself pinned.
Also, I suspect most of the problematic meetings are from outlook but for whatever reason aren't registered as 'Teams' meetings (if someone just pasted the link into the invite body). An easy workaround would be to go to Teams "Calendar" tab, double-click to open the meeting and click on the link to join. At least you don't have to leave the app and do copy-paste this way.

**Comment by jmergy** [comment]
This is working now for me too on Vivaldi (Chrome-based) with the Redirector extension and the above config on Debian 12.1 (Wayland/Gnome)

![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/2380410/4cf3ddff-cbc8-4959-90c4-1fd40f4227e2)

![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/2380410/f0444049-e989-428a-8da9-bca14631bf60)

**Comment by ogghi** [comment]
> This is working now for me too on Vivaldi (Chrome-based) with the Redirector extension and the above config on Debian 12.1 (Wayland/Gnome) so now I can click on invites with teams.microsoft.com URLs - it kicks to Vivaldi, then to teams-for-linux if I decide. Cool!
> 
> ![image](https://private-user-images.githubusercontent.com/2380410/258232159-4cf3ddff-cbc8-4959-90c4-1fd40f4227e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDYwNjEsIm5iZiI6MTczMjEwNTc2MSwicGF0aCI6Ii8yMzgwNDEwLzI1ODIzMjE1OS00Y2YzZGRmZi1jYmM4LTQ5NTktOTBjNC0xZmQ0MGY0MjI3ZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMTIyOTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDNkOGJlYjYxYzI3ODhhYzk0YTBkOTc3NmFmMzgzODk0Y2ZkYjNmNTNmZDgxOGZmMDk1M2E0YTg4NDQwN2YzOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ob6zFQdOFGshQo5EhIrb23a75Y-Ts6sorRxWXbFCHS8)
> 
> ![image](https://private-user-images.githubusercontent.com/2380410/258232048-f0444049-e989-428a-8da9-bca14631bf60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDYwNjEsIm5iZiI6MTczMjEwNTc2MSwicGF0aCI6Ii8yMzgwNDEwLzI1ODIzMjA0OC1mMDQ0NDA0OS1lOTg5LTQyOGEtOGRhOS1iY2ExNDYzMWJmNjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMTIyOTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjEzNzg0MjA0NmQzYWMyNjMzMjQ5MGQyMzNmMjA5ZTViNjA0YzJiZGU2Y2M4MTUxYzNjZDUzY2I4MDE1YmIyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fl8SZNZNLbKVVkhDHhfazPtTQAdx5jilFG45rrXFxfs)

Wow, this actually WORKS! :+1:  :100:

**Comment by houzefa-abba** [comment]
> This is working now for me too on Vivaldi (Chrome-based) with the Redirector extension and the above config on Debian 12.1 (Wayland/Gnome) so now I can click on invites with teams.microsoft.com URLs - it kicks to Vivaldi, then to teams-for-linux if I decide. Cool!
> 
> ![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/2380410/4cf3ddff-cbc8-4959-90c4-1fd40f4227e2)
> 
> ![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/2380410/f0444049-e989-428a-8da9-bca14631bf60)

Thanks works fine! Here is my version where I keep msteams as a full scheme (with `://`) then re-include teams.microsoft.com:
![Image](https://github.com/user-attachments/assets/c0de1ba2-9fd4-41e0-bc5d-e562dec875b6)

For folks missing the last part (making your system handle msteams links), I suggest grepping though `.local/share/applications` - in my case:
* I had a `.local/share/applications/com.microsoft.Teams.desktop` remnant from my previous install of that (now deprecated) flatpak; I edited inside to reference this flatpak & updated the `--command=` flag. Attaching it below, if it helps.
* in `.local/share/applications/mimeinfo.cache` I have: `x-scheme-handler/msteams=com.microsoft.Teams.desktop;`

`.local/share/applications/com.microsoft.Teams.desktop` (had to rename it for github comments):
[com.microsoft.Teams.desktop-rename-me.txt](https://github.com/user-attachments/files/18735735/com.microsoft.Teams.desktop-rename-me.txt)

**Comment by mirenradia** [comment]
Recently, this has stopped working for me. Redirector successfully redirects the link but teams-for-linux just refreshes (as if I had pressed Ctrl+R) and doesn't open the meeting.

**Comment by IsmaelMartinez** [comment]
@mirenradia , is that with the latest pre-release or are you using a stable release? Thanks

**Comment by mirenradia** [comment]
> [@mirenradia](https://github.com/mirenradia) , is that with the latest pre-release or are you using a stable release? Thanks

I'm using 1.12.7 from the RPM repo so the stable release.

**Comment by IsmaelMartinez** [comment]
oh, can you try the pre-release one or the previous version? i think that one had a wee bugger https://github.com/IsmaelMartinez/teams-for-linux/issues/1573 so either should fix your issue (hopefully)

**Comment by mirenradia** [comment]
> oh, can you try the pre-release one or the previous version? i think that one had a wee bugger [#1573](https://github.com/IsmaelMartinez/teams-for-linux/issues/1573) so either should fix your issue (hopefully)

I tried installing 1.12.8 but that hasn't resolved the problem for me.

**Comment by IsmaelMartinez** [comment]
@mirenradia , can you open an issue with as much info as possible. 

Also, worth trying with a couple of previous versions, just to see if it is something added in the last stable one. 

That way is a bit easier to understand what the problem is. 
Thanks!

**Comment by mirenradia** [comment]
OK, I reread the above messages again and it turns out my redirector config was using the old "v1" format suggested by @jmergy. Switching to the "v2" format (with the extra `teams.microsoft.com` bit) suggested by @houzefa-abba resolves the issue for me and teams-for-linux opens meetings successfully.

Sorry for the trouble.

**Comment by dnet890** [comment]
@mirenradia I tried to configure redirector and it does not worked

<img width="726" alt="Image" src="https://github.com/user-attachments/assets/cd914de8-d61d-4125-9550-61039da46189" />

**Comment by IsmaelMartinez** [comment]
Hi @dnet890, that pattern looks a bit different than the one @houzefa-abba provided. 

<img width="677" alt="Image" src="https://github.com/user-attachments/assets/db72d741-952f-48ba-9dd6-b31b5c65f548" />

Hope heps

**Comment by dnet890** [comment]
Hi
the extension accept the parameters. But, it cannot redirect me to the deb version of the app. What do you think?

<img width="778" alt="Image" src="https://github.com/user-attachments/assets/5fb59743-cf68-4d28-8f67-ebd3152ac632" />

**Comment by houzefa-abba** [comment]
@dnet890 I suggest checking into `.local/share/applications` with tips I noted in my message above : https://github.com/IsmaelMartinez/teams-for-linux/issues/675#issuecomment-2648312616

Hope it helps

**Comment by dnet890** [comment]
Hi
I use deb file and I cannot find the file in /.local/share/applications/. do you mean from /usr/share/applications/?

<img width="459" alt="Image" src="https://github.com/user-attachments/assets/42e946ac-152a-421f-820e-cb6f7d7ed72f" />

**Comment by houzefa-abba** [comment]
Yeah probably; in my case I installed from flatpak: https://flathub.org/apps/com.github.IsmaelMartinez.teams_for_linux

Comparing your file with the one I pasted in <https://github.com/IsmaelMartinez/teams-for-linux/issues/675#issuecomment-2648312616>, the `Exec` part is different; in particular, mine contains `@@u %U @@` at the end which forwards arguments like that msteams link

**Comment by dnet890** [comment]
I tried this parameter for deb file. But, it still redirect me for web version

<img width="957" alt="Image" src="https://github.com/user-attachments/assets/4fb6a232-e13e-470e-9ceb-56b10279347f" />

**Comment by hightowe** [comment]
Since people are still commenting on this tread with struggles, I decided to post my solution from when this thread was first started, well over two years ago. It has worked the entire time, and still does. I use it on Linux Mint. 

```
hightowe@eden ~ $ cat ~/bin/teams4linux 
#!/bin/bash

###############################################################
# Quick xdg-open launcher script that was inspired by
# https://200ok.ch/posts/2023-02-07_make_teams_work_again.html
# which I could not get to work on my bash...
#
# Written 2023-Feb-20 by Lester Hightower
#
# NOTES:
# $ xdg-mime query default x-scheme-handler/msteams
# teams-for-linux.desktop
#
# https://askubuntu.com/questions/18418/how-can-i-set-which-application-is-launched-by-xdg-open
###############################################################

# The BASE_URL to replace with msteams:
BASE_URL='https://teams.microsoft.com'

url="$1"

echo "$url" | egrep -q -e "^$BASE_URL"
if [ $? -ne 0 ]; then
  echo "Invalid url vs expected ($BASE_URL): $url"
  exit -1
fi

path=$(echo "$url" | sed -e  "s%^$BASE_URL%%")
if [ $? -ne 0 ]; then
  echo "Failed to find path with sed"
  exit -1
fi

new_url="msteams:$path"

echo -e "RUNNING:\n"xdg-open "'$new_url'"

# Uncomment the next line if you want xdg-open to be run
#xdg-open "$new_url"
```

**Comment by dnet890** [comment]
it looks like it does not worked because I think this is for old discontinued MS Teams app

<img width="428" alt="Image" src="https://github.com/user-attachments/assets/7cc50ced-dcac-4456-b45e-4da96e9ca405" />

**Comment by IsmaelMartinez** [comment]
@dnet890 , that is the one microsoft supported. Either uninstall it and use the one in this repo, or see if MS can help you out (but I doubt as it was discontinued a few years ago)

**Comment by dnet890** [comment]
Yes, It detect the script when I use the script it does not worked out and they still force me to use the web version. 

<img width="490" alt="Image" src="https://github.com/user-attachments/assets/77e695ca-4a3a-4135-9c6b-8b6f99712282" />

[data.zip](https://github.com/user-attachments/files/20769429/data.zip)

**Comment by hightowe** [comment]
> Yes, It detect the script when I use the script it does not worked out and they still force me to use the web version.
> 
> <img alt="Image" width="490" src="https://private-user-images.githubusercontent.com/29157187/455846366-77e695ca-4a3a-4135-9c6b-8b6f99712282.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNjIxNDcsIm5iZiI6MTc1MDE2MTg0NywicGF0aCI6Ii8yOTE1NzE4Ny80NTU4NDYzNjYtNzdlNjk1Y2EtNGEzYS00MTM1LTljNmItOGI2Zjk5NzEyMjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE3VDEyMDQwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiOGNhN2RhNDBhMmQ4ZDg0ZTNkNzdjNmI0MWZiZjYxMGQ1ZjY5NmY3NjkwMDM5Yzk2YmQ5MzdkNWJkNmJjZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.44LmZjDhSCgcG2M74II6w41X2tryO9VhDfmbxasaz8Q">
> [data.zip](https://github.com/user-attachments/files/20769429/data.zip)

1. The first line in what I shared (the cat command) is not part of the program, but you included it.
2. You did not give the program a URL to operate on.

**Comment by dnet890** [comment]
Yeah, I can make it work with the combination of redirector extension ([the script](https://github.com/IsmaelMartinez/teams-for-linux/issues/675#issuecomment-2975525983)) and the sh script from you  (https://github.com/IsmaelMartinez/teams-for-linux/issues/675#issuecomment-2976316826). it is quite roundabout setup compared to Windows version. But, I hope in the future we do not need this complicated setup to make it easier to join meetings.

**Comment by victornoel** [comment]
Hi, I'm struggling also with this, I tried many combinations of everything shared, and the closed I got to it was to call manually in a terminal the following:
```sh
teams-for-linux-electron "https://teams.microsoft.com/l/message/19...."
```

It will reload the opened teams window from zero and then bring me to the expected conversation.

But anything else and even when setting the redirect pattern to `msteams:https://teams.microsoft.com/$1` (for the pattern wildcard `https://teams.microsoft.com/*`) does not work: I can see xdg-open is properly called and teams-for-linux-electron is properly called and opened but nothing happens in there (apart from the mouse cursor being moved on top of it by Gnome I suppose).

So I can't even manage to reproduce the same behaviour as calling teams-for-linux-electron in the shell manually as I show in the snippet above. The desktop file is the one shipped in the rpm I believe, cf https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=teams-for-linux-electron-bin.

**Comment by IsmaelMartinez** [comment]
hi @victornoel , this one is a tricky one as it depends on the package system you are using and your os configuration. It works fine for a lot of people, but for some (many) we get different behaviors. 

I do think is something to do with the xdg-open configuration. There might be an old configuration kicking about and it gets a bit confused on where to open the app, or it muddles it a bit too much. 

It is also important to notice that if you are clicking the link withing the app, doing ctrl+click might be "a solution" as that will open the url within the app.

Sorry I can't help much, and thanks for sharing

**Comment by victornoel** [comment]
@IsmaelMartinez thx and maybe you could confirm something for me: what is the exact parameter that teams is expecting to get on the command line for this to work? is it `"https://teams.microsoft.com/l/message/19...."` as I mentioned? Something else?

Because this is passed as-is to the binary from this project, so at least this shouldn't vary between packaging versions :) it will be helpful to know this, I don't think it was explicitly shared here yet.

**Comment by IsmaelMartinez** [comment]
either the https://teams.microsoft.com/l/message/... or the msteams one should work. 

It uses the regex from here https://github.com/IsmaelMartinez/teams-for-linux/blob/74627c351b49fad8ce92db929d24abbafb780201/app/config/README.md?plain=1#L57 (that you can configure if you want to try something else)

**Comment by victornoel** [comment]
@IsmaelMartinez this could explain why what I tried didn't work, my links does not match any of the regexp used in the code and the one I can provide with `meetupJoinRegEx` does not extract the correct url :)

I will try to summarize my findings:
- this is the code that deals with this: https://github.com/IsmaelMartinez/teams-for-linux/blob/52f7f6d76186b81fe71d779a5f9315870722e173/app/mainAppWindow/index.js#L221-L248
- it does 3 things
  - check the `meetupJoinRegEx` and use the argument as-is
  - else check `v1msTeams` and replaces `msteams:/` with the correct url from the config
  - else check `v2msTeams` and replaces `msteams` with `https`
- so my understanding is that we need a real https url for this to work
- `meetupJoinRegEx` can never work if we are coming from xdg-open or firefox handling of `msteams:/` links (after rewriting by Redirector for example) as it does not strip the `msteams:/` prefix of the url being passed to it (at least on my system…)
- also none of the regex match my url, it contains /l/message, but none of the 3 options specified in the regex

IMO the solution is thus:
- to add support /l/message on top of the others
- and maybe also add way to extract the url from the `meetupJoinRegEx` (if that's even possible…)

WDYT @IsmaelMartinez? Am I missing something or does it track?

**Comment by victornoel** [comment]
Another thing I noticed is:
- if I replace message by channel in my link of the form `msteams://teams.microsoft.com/l/channel.....` then it works
- but it reload the whole page instead of just moving to it so I'm not sure that's what we expect?

**Comment by IsmaelMartinez** [comment]
oh, you shoudl be albe to modify the config "meetupJoinRegEx" config to handle then the message ones. 

```
{
   "meetupJoinRegEx": "^https://teams.(microsoft|live).com/.*(?:meetup-join|channel|chat|message)"
}
```

Let me know if that works for you. If it does then we can modify those lines.

Thanks for the testing!

**Comment by victornoel** [comment]
@IsmaelMartinez I cannot use that regex because with `xdg-open`, the url MUST start with `msteams:` to work

**Comment by victornoel** [comment]
@IsmaelMartinez can't we just either add `message` in the last capture group of all the regex in the code or maybe even better remove that capture group and just match `^msteams:\/\/teams\.microsoft\.com\/l\//`?

**Comment by IsmaelMartinez** [comment]
hey @victornoel , I add that as a config option (like the other regex) and add the message in there. If that doesn't work at least it gives you the option to play with the msteams message.

I will try to make it happen tonight, but it might be tomorrow. Apologies for the delay.

**Comment by IsmaelMartinez** [comment]
Can you try the pre-release https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v2.1.4 ? 

The config option looks like this (defaults):

```
{
  ...
  msTeamsProtocols: {
       "v1": "^msteams:/l/(?:meetup-join|channel|chat|message)",
       "v2": "^msteams://teams.microsoft.com/l/(?:meetup-join|channel|chat|message)"
  },
  ...
}
```

Play around and let me know if this fixes your issue (and if any change is needed in the defaults)

Some config examples can be found in https://github.com/IsmaelMartinez/teams-for-linux/blob/main/docs/configuration.md 

Hopefully this solves this issue. Thanks for all the investigation. I really appreciate it.

**Comment by victornoel** [comment]
@IsmaelMartinez so it kind of works out of the box now with the following Redirector settings:

<img width="709" height="337" alt="Image" src="https://github.com/user-attachments/assets/00a8262a-c805-48e0-9303-ee2e8f0d3b0a" />

The only potential problem is that it will reload the whole UI instead of just redirecting, but that's acceptable. I'm curious to know if that can be improved, but how? :)

If I use instead the v1 pattern (`msteams:/l/$1`) then the page reload but it does not redirect me to the right place, just to the same page I was before 🤷

**Comment by IsmaelMartinez** [comment]
nice! Well, I guess we can finally close this. Maybe we can improve it in the future but, if it works, it works. Thanks for checking and finding the reason why it didn't work.
